### PR TITLE
feature(f.promisify): Add F.Promisify type

### PR DESCRIPTION
**Pull request by Andarist**
## 🎁 Pull Request

* [x] Used a clear / meaningful title for this pull request
* [x] Tested the changes in your own code (on your projects)
* [x] Added / Edited tests to reflect changes (`tst` folder)
* [x] Have read the **Contributing** part of the **Readme**
* [x] Passed `npm test`

Gonna add tests once you give me a green light on this one - I've created this PR rather quickly, but  looks that understanding how tests are actually written might take a little bit more time, so I've postponed it for now.

#### Why have you made changes?

It was a useful type that I have used in my project and thought it might find its home here as well.

#### Is there any breaking changes?

* [ ] Yes, I changed the public API & documented it
* [ ] Yes, I changed existing tests
* [x] No,  I added to the public API & documented it
* [ ] No,  I added to the existing tests
* [ ] I don't know

### CODE REVIEW

**Comment by Andarist** [comment]
Cool, I'll add tests - might take me a few days as I'm pretty busy at the moment, gonna ping you when I'm done. Cheers!

**Comment by Andarist** [comment]
@pirix-gh I've added test + changed the implementation to handle promise flattening, which kinda should be handled by TS on its own, but unfortunately it is not and the issue about it is in their tracker

File: src/Function/Promisify.ts [hunk 1]
```diff
@@ -12,4 +12,6 @@ import {Return} from './Return'
  * type test0 = F.Promisify<(a: number) => number> // (a: number) => Promise<number>
  * ```
  */
-export type Promisify<F extends Function> = (...args: Parameters<F>) => Promise<Return<F>>
+export type Promisify<F extends Function> = Return<F> extends Promise<any>
```
Comment (pirix-gh) [inline src/Function/Promisify.ts]: Nice

**Comment by pirix-gh** [comment]
:tada: 

Thanks for your contribution!

**Pull request merged**